joomla/joomla-cms
# Refactory com_banners tracks export modal + new toolbar button layout

Pull Request for Issue #10911 & PR #10927 (improve it).
#### Summary of Changes

**Download Tracks Modal:**
- Fix and improve HTML rendering (use <code>renderField</code>)
- No tooltip for File Name, as description is a bit long (and not always easy to use tooltip when long description).
- New form field type <code>note</code> for <code>File Name</code> description in a alert-info style.
- Tooltip placement bottom for <code>Compressed</code> label (prevent tooltip cropped at top when in iframe)
- Buttons in the BS modal footer
- Using new viewport dimensions
- **HATHOR** : modal was broken! This PR fixes it in Hathor Template, and add overrides to adapt rendering.

**New Generic Toolbar Button Layout:**
- appendButton using a new toolbar button layout:

```
$selector = $displayData['selector'];
$class    = isset($displayData['class']) ? $displayData['class'] : 'btn btn-small';
$icon     = isset($displayData['icon']) ? $displayData['icon'] : 'out-3';
$text     = isset($displayData['text']) ? $displayData['text'] : '';
?>
<button class="<?php echo $class; ?>" data-toggle="modal" data-target="#<?php echo $selector; ?>">
    <span class="icon-<?php echo $icon; ?>"></span>
    <?php echo $text; ?>
</button>
```
#### Testing Instructions

To be tested on staging or 3.6.0-beta
- Test on Isis template (api) and Hathor template (overrides)
- Go to **Components** > **Banners** > **Tracks**
- Click on <code>Export</code> toolbar button
- Check tooltip placement
- Test <code>Cancel</code> and <code>Export</code> buttons now in modal footer

**Isis admin template**
Before:
![capture d ecran 2016-06-26 a 20 42 47](https://cloud.githubusercontent.com/assets/2385058/16364073/1a610f04-3bdf-11e6-86aa-9f66ba9b9d57.png)

After:
![capture d ecran 2016-06-25 a 17 04 16](https://cloud.githubusercontent.com/assets/2385058/16364078/34a8a91c-3bdf-11e6-9e09-080e20db9e18.png)

**Hathor admin template**
Before:
![capture d ecran 2016-06-26 a 20 37 30](https://cloud.githubusercontent.com/assets/2385058/16364082/461f2400-3bdf-11e6-866a-0743c18dcd52.png)

After:
![capture d ecran 2016-06-26 a 20 38 40](https://cloud.githubusercontent.com/assets/2385058/16364085/50f5a1ec-3bdf-11e6-9a46-454187f102fc.png)
